docs/matrix.md

```diff
@@ -172,6 +176,98 @@ The `Matrix.Include` will take `Parameters` of type `"string"` only.
   ...
 ```
 
+#### Include specific combinations in the Matrix
+
+In the example below, `Pipeline` will need to define a `Matrix` that will execute golang-test in three different linux flavors.
+
+While running on amd64 and ppc64le, golang-test should execute `go test github.com/tektoncd/pipeline/pkg/... -race -cover -v`.
+While running on linux/s390x, golang-test should execute `go test github.com/tektoncd/pipeline/pkg/... -cover -v`.
+
+```yaml
+apiVersion: tekton.dev/v1beta1
+kind: Pipeline
+metadata:
+  name: pipeline-to-build-and-test-go-project
+spec:
+  workspaces:
+    - name: shared-workspace
+  tasks:
+    - ...
+    - name: golang-test
+      taskRef:
+        name: golang-test
+      workspaces:
+        - name: source
+          workspace: shared-workspace
+      params:
+        - name: package
+          value: "github.com/tektoncd/pipeline"
+        - name: packages
+          value: "./pkg/..."
+      matrix:
+        params:
+          - name: GOARCH
+            value:
+              - "linux/amd64"
+              - "linux/ppc64le"
+              - "linux/s390x"
+        include:
```

Comment:
TEP-0118 spells out a really clear example that covers a lot of different cases. We don't necessarily have to put all those cases in one example, but all the cases should be covered in the docs.

Reply:
Added the additional use cases.